### PR TITLE
refactor(*): switch to typescript

### PR DESCRIPTION
**Pull request by bnjjj**

### CODE REVIEW

File: bots/messageTypes/postback/feedback.ts [hunk 1]
```diff
@@ -1,28 +1,30 @@
...
 });
-const Message = require("../../../models/messages.model");
-const { TextMessage } = require("../../../platforms/generics");
 
-module.exports = [
+export default [
   {
     regx: "FEEDBACK_GOOD_([^_]*)_(.*)",
-    action (senderId, postback, regx) {
+    action (senderId, postback, regx, entities?, res?): Promise<any> {
```
Comment (makidelille) [inline bots/messageTypes/postback/feedback.ts:15]: "entities?, res? " if it's for param consistency, why not do the same for the messageTypes ? :)

File: bots/utils.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+import * as URL from "url";
+const config = cfg.load();
+
+export function getOvhClient (senderId): Promise<any> {
+  return User.findOne({ senderId }).exec().then((userInfos: any) => {
+    if (!userInfos) {
+      return Bluebird.reject({ statusCode: 403, message: "Tu dois d'abord te connecter. Pour ce faire tu peux me le demander" });
+    }
+
+    const ovhClient = ovh({
```
Comment (makidelille) [inline bots/utils.ts]: return ovh({...}):

File: config/middlewares/verifyRequest.ts [hunk 3]
```diff
@@ -1,14 +1,14 @@
...
+import {createHmac} from "crypto";
 
 //  * Verify that the callback came from Facebook. Using the App Secret from
 //  * the App Dashboard, we can verify the signature that is sent with each
 //  * callback in the x-hub-signature field, located in the header.
 //  *
 //  * https://developers.facebook.com/docs/graph-api/webhooks#setup
 //  *
-module.exports = (config) => function verifyRequestSignature (req, res, buf) {
+export default (config?) => function verifyRequestSignature (req, res, buf) {
```
Comment (makidelille) [inline config/middlewares/verifyRequest.ts]: why config optional ?

**Review by ghost — APPROVED** [review]

File: platforms/generics.ts [hunk 4]
```diff
@@ -34,9 +43,14 @@ class ButtonsMessage {
   }
 }
 
-class ButtonsListMessage {
+export class ButtonsListMessage {
+  text: string;
+  delete_original: boolean;
```
Comment (makidelille) [inline platforms/generics.ts:48]: useless ?

**Review by makidelille — COMMENTED** [review]
some questions :)

File: bots/messageTypes/postback/hosting.ts [hunk 5]
```diff
@@ -59,7 +61,8 @@ module.exports = [
 
           if (err.error === 460) {
             return Bluebird.resolve([
-              new TextMessage("Ton service hébergement web semble être suspendu, pour le réactiver il faut le renouveler via le manager", `Voici un guide qui va te permettre de renouveler ton hébergement web : ${guides.renewOvh}`)
+              new TextMessage("Ton service hébergement web semble être suspendu, pour le réactiver il faut le renouveler via le manager"),
```
Comment (antleblanc) [inline bots/messageTypes/postback/hosting.ts]: `via le manager` => `via l’espace client` 🍭

File: bots/messageTypes/postback/feedback.ts (hunk 1 above)
Comment (bnjjj) [reply bots/messageTypes/postback/feedback.ts:15]: done

File: platforms/generics.ts (hunk 4 above)
Comment (bnjjj) [reply platforms/generics.ts:48]: nope

**Review by makidelille — APPROVED** [review]

**Pull request closed without merging**